src/main/java/seedu/address/model/project/UniqueProjectList.java

```diff
@@ -0,0 +1,142 @@
+package seedu.address.model.project;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.project.exceptions.DuplicateProjectException;
+import seedu.address.model.project.exceptions.ProjectNotFoundException;
+
+/**
+ * A list of projects that enforces uniqueness between its elements and does not allow nulls.
+ * A project is considered unique by comparing using {@code Project#isSameProject(Project)}. As such, adding and
+ * updating of projects uses Project#isSameProject(Project) for equality so as to ensure that the project being added
+ * or updated is unique in terms of identity in the UniqueProjectList. However, the removal of a project uses
+ * Project#equals(Object) so as to ensure that the project with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Project#isSameProject(Project)
+ */
+public class UniqueProjectList implements Iterable<Project> {
+
+    private final ObservableList<Project> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Project> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent project as the given argument.
+     */
+    public boolean contains(Project toCheck) {
+        requireNonNull(toCheck);
+
+        return internalList.stream().anyMatch(toCheck::isSameProject);
+    }
+
+    /**
+     * Adds a project to the list.
+     * The project must not already exist in the list.
+     */
+    public void add(Project toAdd) {
+        requireNonNull(toAdd);
+
+        if (contains(toAdd)) {
+            throw new DuplicateProjectException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the project {@code target} in the list with {@code editedProject}.
+     * {@code target} must exist in the list.
+     * The project identity of {@code editedProject} must not be the same as another existing project in the list.
+     */
+    public void setProject(Project target, Project editedProject) {
+        requireAllNonNull(target, editedProject);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new ProjectNotFoundException();
+        }
+
+        if (!target.isSameProject(editedProject) && contains(editedProject)) {
```

Comment:
Should it be target.isSameProject(edited) && contains(editedProject) without the ! in front of target.isSameProject(edited)?

Reply:
can confirm no issue here, if target and edited are not the same, and current list already contains new edited project, then the project is duplicated.